libs/state/src/lib/download-file/download-file.state.ts

```diff
@@ -0,0 +1,112 @@
+import { Action, State, StateContext } from '@ngxs/store';
+import { jsPDF } from 'jspdf';
+import { DownloadFile, FileFormat } from './download-file.action';
+
+//TODO: For testing purpose only
+const fileUrl = 'assets/compass.svg';
+
+/**
+ * Download state model
+ */
+export interface DownloadStateModel {
+  file: File | null;
+}
+
+/**
+ * Download State Model used to convert
+ * file from SVG file content to different
+ * format and download to user.
+ */
+@State<DownloadStateModel>({
+  name: 'download',
+})
+export class DownloadState {
+  /**
+   * Download converted formatted file to browser
+   * @param blob
+   * @param fileName
+   */
+  blobDownload(blob: Blob, fileName: string) {
+    const url = window.URL.createObjectURL(blob);
+    const anchor = document.createElement('a');
+    document.body.appendChild(anchor);
+    anchor.download = fileName;
+    anchor.href = url;
+    anchor.click();
+    anchor.remove();
+    window.URL.revokeObjectURL(url);
+  }
+
+  /**
+   * Actions download file in specified format
+   * @param ctx
+   * @param action
+   */
+  @Action(DownloadFile)
+  async downloadFile(ctx: StateContext<DownloadStateModel>, action: DownloadFile) {
+    const fileName = fileUrl.substring(fileUrl.lastIndexOf('/') + 1, fileUrl.lastIndexOf('.'));
+
```

Comment:
If url is available then download directly. Otherwise if format is supported convert